frontend/src/pages/start/main/Main.tsx

```diff
@@ -42,35 +71,181 @@ const Main: FunctionComponent = () => {
     history.push(`/${routes.start}/configure`);
   }, []);
 
+  const onDelete = async (project: Project) => {
+    try {
+      splitbee.track("Configuration: delete");
+      await projectService.removeProject(project.id);
+      toast(`Project "${project.name}" deleted!`);
+    } catch (e) {
+      toast.error("Something went wrong: can not delete custom emulator");
+    } finally {
+      closeDialog();
+    }
+  };
+
+  const closeDialog = () => {
+    setShowDialog(null);
+  };
+
+  const openDialog = (type: DialogTypes, props?: any) => {
+    // return is just to break out of switch
+    switch (type) {
+      case DialogTypes.deleteProject:
+        return setShowDialog({
+          body: (
+            <>
+              <h3>Delete project</h3>
+              <span>Are you sure you want to delete this project?</span>
+            </>
+          ),
+          onConfirm: () => onDelete(props),
+          confirmBtnLabel: "DELETE",
+          cancelBtnLabel: "BACK",
+        });
+      case DialogTypes.openProject:
+        return setShowDialog({
+          body: (
+            <>
+              <h3>New emulator</h3>
+              <span>Not supported yet :(</span>
+            </>
+          ),
+          confirmBtnLabel: "CREATE",
+          cancelBtnLabel: "CANCEL",
+        });
+    }
+  };
+
+  const tabsDom = {
+    // Projects tab
+    [tabs[0].id]:
```

Comment:
This is nice, but I think it would be cleaner to put the tab UI in the `tabs` array itself. Because in that way all the related code is in a single place and you don't need to reference to which tab it belongs by an array index - which could change. There is no way to inforce the developer to change the `tabsDom` array when `tabs` array indexes change.

https://github.com/onflowser/flowser/blob/c63ed5199cc532adbc6b7cef1f7e76f3a71c13db/frontend/src/pages/start/main/Main.tsx#L28-L31

Maybe something like this?

```tsx
const tabs = [
  {
    id: "projects",
    label: "Projects",
    content: <div>The content itself...</div>,
    default: true,
  },
  // ...
];
```

Reply:
Yes sure, this is better.